### ISSUE TITLE
A few points of Clarification

### ISSUE THREAD

**Issue by mbarszcz-pcb** [tebl/BulkyMIDI-32#2]
Tebl, 

First of all, thank you for this project.  It is nice to have a project that makes the MT32-Pi feel like a complete device, rather than just a While I was building it, I took down a few notes and I would like to provide some constructive feedback that I think would help clear things up for people, and improve the design even further.

1. The tactile buttons on the front panel should be 12mm.  I didn't find this indicated anywhere in the build documentation just that they were "slightly long" and it took me some trial and error.  Specifically the part # I found that worked well for my was Digikey # [TS02-66-120-BK-160-LCR-D](https://www.digikey.com/en/products/detail/cui-devices/TS02-66-120-BK-160-LCR-D/15634353).

2.  I'd recommend the diodes on the extras board should be 1N5819 Schottky Diodes instead of 1N4001s standard silicon diodes.  The Vf of the 1N4001 is 0.7V, which means the 5V rail on the extras board is down to only 4.3v-4.2v.  It does work, but this is getting pretty low for a 5V rail.  A 1N5189 is the same price and only has a 0.3V drop.  It would be a drop in replacement.

3.  I wasn't sure what the "faceplate" connector J4 was for on the main board.  I looked and looked, but I couldn't figure out a purpose for it.  I wound up ordering extra connectors for it as part of the BOM that ultimately never got used.  Perhaps it used to be part of an older revision and no longer serves a purpose.  Perhaps this space could be repurposed for #4.

4.  Since the extras board is really designed to be used in conjunction with the main board, the power socket on it is redundant, but I think this space could be used to add a convenient power switch..  What would be nice to have here instead would be a PCB mount, right angle toggle switch to turn the whole bulkymidi on and off.  Perhaps a 2 pin header (in place of J4) could pass the power down from the DC jack on the main board down to the extras board, to the switch, and then back up to the main board.  This could be bypassed with a solder jumper on the main board if the extras board was not used.  I wired mine up by hand (photo attached), but perhaps it could be incorporated into the design.

![rear-power-switch](https://user-images.githubusercontent.com/62184001/203842008-adb3e6b0-574b-4e4e-8c38-09935de26872.jpg)

5. BOM suggestions:
    a. extas - I found no use for the M2 screws/nuts on the Extras board.  Either I miseed what these are for, or they are no longer needed in the current revision.
    b.  The knobs on the extras board can be no larger than ~15mm in diameter.  I learned this the hard way.
    c.  Main board - J4 was not needed
    d.  A note on the MIDI jacks (5 pin DIN).  There seem to be two distinct styles of these.  One style with the two front pins close together at 5mm [Example - Digikey P/N CP-2350-ND ](https://www.digikey.com/en/products/detail/cui-devices/SDS-50J/97033)(the more common) , and another style with the front two pins further apart at 10mm (what the PCB uses, and common on AliExpress) - [Example Digikey P/N 1939-G-1006-ND](https://www.digikey.com/en/products/detail/globtek-inc/G-1006/16515764?s=N4IgTCBcDaIIwE4DMCC0BxVcAM2BsqAcgCIgC6AvkA).  You may want to indicate this on the BOM so it is clear which part is needed.  Or alternately, add holes at 5mm (in addition to the 10mm) on the PCB so either style would be compatible.

6. A few additional pictures of the sides/back of a completed MT-32 Pi would be helpful during the build process to see how it all stacks together if you could include them, especially the way the raspberry pi can mount on the standoffs and the header.
![raspberry-pi-mounting](https://user-images.githubusercontent.com/62184001/203842006-c31969c0-0ed8-4dfc-affc-388fa0cbbbdc.jpg)

Thanks again,
Matt

**Comment by tebl** [comment]
Thank you for the detailed feedback, all valid points that I will have to consider on how to integrate into things. A lot of the time I'm ordering parts, receiving them two months later without any discernible markings so different sizes ordered just in case tend to blend together. It is fun to see that the prototype I built for myself isn't the only one in existence :-)

**Comment by hpingel** [comment]
So, for boasting with pictures of my own BulkyMIDI build (ongoing), should I use this ticket here or should I open my own ticket or is there a user gallery somewhere to show off tebl projects that I built at home?

I will just post one pic here and patiently wait for further instructions. :-)
![IMG_20230116_113057_HDR](https://user-images.githubusercontent.com/646865/213280330-ff4fd555-1065-40e3-9874-b6dde7168592.jpg)

**Comment by tebl** [comment]
As long as you're OK with it being used as part of the project, just post
them here and I'll add them to a section of the gallery 😀

ons. 18. jan. 2023, 20:50 skrev hpingel ***@***.***>:

**Comment by hpingel** [comment]
Yes, you can use the picture(s). I have the situation that envelopes with parts arrive every second day and I‘m always thinking: Wait with taking pictures until those parts have arrived and are soldered in…

I just read your FAQ and maybe I will do a small video on my YT Channel instead of pictures. Let‘s see. Thank you anyway for sharing your projects!

**Comment by hpingel** [comment]
Yesterday night I added more components to the Extras board and the passive mixer is now complete. I find this passive mixer really handy. I have a "big" mixer but the number of its channels are limited (3 stereo channels) and the passive mixer in BulkyMIDI-32 now allows me to merge some of the channels before passing the result into my big mixer. This is great. To describe a use case: I have soldered myself a DigiMax Dual extension for the C64 and this extension has two stereo line-outs which can go into the passive mixer of BulkyMIDI when needed.

On the main PCB I have replaced the resistor in position R7 to something higher as I believe it is related to the power LED. My LED appeared too bright and the other LEDs on the Extension board also have 2K2 ballast resistor with them. In the documentation it would be nice to clarify that R7 works as a ballast resistor as it is a bit tricky to trace it to the front panel.

The MIDI-Out/Thru stuff on the Extras board (pink color code) is something I would like to use but I try to understand the routing options, also in conjunction with the routing options in the software of @dwhinham . I came accross the MIDI-Routing branch in his git repository and try to think of possible options. 

Basically, MIDI-Thru on the Extras PCB puts out what comes in via MIDI-In (on the main PCB MIDI-In or through the MIDI-In options on the Extras board, don't use both at the same time?).

And MIDI-Out on the Extras board puts out what the software decides to put out on the Tx pin of the Pi. Which is normally nothing I suppose. It can be configured in software to act as MIDI-Thru (which would duplicate the MIDI-Thru mentioned above) or it could be used for more complex routing purposes via the options in the midi-routing branch (https://github.com/dwhinham/mt32-pi/tree/midi-routing). So a theoretical example scenario could be:
MIDI source #1 goes in via MIDI-In to the Rx pin of the Pi and is audible through the active synth emulation on the Pi.
MIDI source #2 goes into the Pi via USB MIDI connection or network MIDI and is routed to MIDI-Out.
In this scenario one would get the messages of source1 on the MIDI Thru sockets and the messages of MIDI source2 on the MIDI-Out sockets. Don't ask me why one would like to set it up like this - I just try to understand routing options. :)

**Comment by hpingel** [comment]
It's getting more and more bulky!

![Uploading IMG_20230211_180954~2.jpg…]()
